collections/_article/dev-snapshot-godot-4-5-dev-4.md

```diff
@@ -0,0 +1,125 @@
+
+![Stacked outlines](/storage/blog/dev-snapshot-godot-4-5-dev-5/stacked-outlines.webp)
+
+### Specular occlusion from ambient light
+
+Our renderer currently lacks a cheap option for specular occlusion, causing certain metallic/reflective materials to still recieve reflections when they should be darkened/occluded. [Lander](https://github.com/lander-vr) rectifies this limitation in [GH-106145](https://github.com/godotengine/godot/pull/106145), where specular occlusion is added based on ambient light. As users might prefer the old visuals, this is now handled through a toggle in the project settings; the differences can be observed here:
+
+|                                                        Disabled                                                        |                                                       Enabled                                                        |
+| :--------------------------------------------------------------------------------------------------------------------: | :------------------------------------------------------------------------------------------------------------------: |
+| <img src="/storage/blog/dev-snapshot-godot-4-5-dev-5/specular-disabled-1.webp" alt="Specular disabled 1" width="350"/> | <img src="/storage/blog/dev-snapshot-godot-4-5-dev-5/specular-enabled-1.webp" alt="Specular enabled 1" width="350"/> |
+| <img src="/storage/blog/dev-snapshot-godot-4-5-dev-5/specular-disabled-2.webp" alt="Specular disabled 2" width="350"/> | <img src="/storage/blog/dev-snapshot-godot-4-5-dev-5/specular-enabled-2.webp" alt="Specular enabled 2" width="350"/> |
+| <img src="/storage/blog/dev-snapshot-godot-4-5-dev-5/specular-disabled-3.webp" alt="Specular disabled 3" width="350"/> | <img src="/storage/blog/dev-snapshot-godot-4-5-dev-5/specular-enabled-3.webp" alt="Specular enabled 3" width="350"/> |
+
+### macOS: Embedded window support
+
+After Windows and Linux users got to experience the benefits of embedding their windows in the editor, macOS users have understandably been expressing interest in the same coming to their platform. Easier said than done, as this feature is OS-specific and *very* low-level, so implementation requires someone both extremely knowledgeable in a niche area & actually owning the platform in question for proper testing. Thankfully, both of these qualifications were met by [Stuart Carnie](https://github.com/stuartcarnie), who was up to the task of integrating this behemoth in [GH-105884](https://github.com/godotengine/godot/pull/105884). The results should speak for themselves:
+
+<video autoplay loop muted playsinline>
+  <source src="/storage/blog/dev-snapshot-godot-4-5-dev-5/macos-embedded-window.webm?1" type="video/webm">
+</video>
+
```

Comment:
I would add this technical detail as well, as it can describe why some behavior differences are present across platforms:

```suggestion
</video>

The macOS implementation works differently from the Windows and Linux implementations. Since macOS does not allow the kind of window manipulation that Windows and Linux use for game window embedding, macOS uses an inter-process communication approach where the framebuffer is sent from the game process (which performs off-screen rendering) to the editor window. Input is also sent from the editor window to the game process. This approach is more complex, but is also more robust as it doesn't rely on window management tricks that can fall apart in certain edge cases.
```